src/public-lib/lib/proxyPost.ts

```diff
@@ -100,14 +109,20 @@ export function proxyPost(targetObj:any, methodName:string){
                 cacheSize += sizeQueueElement.size;
                 //console.log(`Added entry to cache from ${sizeQueueElement.methodName} of size ${sizeQueueElement.size}. cacheSize is now ${cacheSize}`);
                 // free memory if necessary
-                tryFreeCache();
+                tryFreeCache(apiCacheLimit, log, sentryLog);
             });
         }
         return storeNode[hash];
     }
 }
 
-export function proxyAllPostMethodsOnClient(obj: any, excluded: string[] = []) {
-    const postMethods = Object.getOwnPropertyNames(obj.prototype).filter((methodName) => /UsingPOST$/.test(methodName) && !excluded.includes(methodName));
-    postMethods.forEach((n)=>proxyPost(obj.prototype, n));
+export function proxyAllPostMethodsOnClient(obj: any,
+                                            excluded: string[] = [],
+                                            postMethodNameRegex: RegExp = /UsingPOST$/,
+                                            apiCacheLimit?: number,
+                                            log?: (message: string) => void,
+                                            sentryLog?: (message: string) => void)
```

Comment:
Added bunch of parameters to make this file generic (i.e. independent from global variables)